tests/i18n/override.py

```diff
@@ -0,0 +1,168 @@
+    assert get_lang_from_cookie(request, SUPPORTED_LANGUAGES) is None
+
+
+def test_get_lang_from_http_header(rf):
+    # Test no header.
+    request = rf.get("")
+    assert 'HTTP_ACCEPT_LANGUAGE' not in request.META
+    assert get_lang_from_http_header(request, SUPPORTED_LANGUAGES) is None
+
+    # Test empty header.
+    request = rf.get("", HTTP_ACCEPT_LANGUAGE='')
+    assert get_lang_from_http_header(request, SUPPORTED_LANGUAGES) is None
+
+    # Test header with wildcard.
+    request = rf.get("", HTTP_ACCEPT_LANGUAGE='*')
+    assert get_lang_from_http_header(request, SUPPORTED_LANGUAGES) is None
+
+    # Test header with supported language.
+    request = rf.get("", HTTP_ACCEPT_LANGUAGE='gl')
+    assert get_lang_from_http_header(request, SUPPORTED_LANGUAGES) == 'gl'
+
+    # Test cookie with longer supported language.
+    request = rf.get("", HTTP_ACCEPT_LANGUAGE='en-US')
+    assert get_lang_from_http_header(request, SUPPORTED_LANGUAGES) is None
+
+    # Test header with longer underscore language code for a supported
+    # language.
+    request = rf.get("", HTTP_ACCEPT_LANGUAGE='gl_ES')
+    assert get_lang_from_http_header(request, SUPPORTED_LANGUAGES) is None
+
+    # Test header with longer hyphen language code for a supported language.
+    request = rf.get("", HTTP_ACCEPT_LANGUAGE='fr-FR')
+    assert get_lang_from_http_header(request, SUPPORTED_LANGUAGES) == 'fr'
+
+    # Test header with shorter language code for a supported language.
+    request = rf.get("", HTTP_ACCEPT_LANGUAGE='en')
+    assert get_lang_from_http_header(request, SUPPORTED_LANGUAGES) is None
+
+    # Test header with unsupported language.
+    request = rf.get("", HTTP_ACCEPT_LANGUAGE='FAIL')
+    assert get_lang_from_http_header(request, SUPPORTED_LANGUAGES) is None
+
+
+@pytest.mark.django_db
+def test_get_lang_from_session(rf, client):
+    # Test no session.
+    request = rf.get("")
+    assert not hasattr(request, 'session')
+    assert get_lang_from_session(request, SUPPORTED_LANGUAGES) is None
+
+    # Test session with no language.
+    response = client.get("")
+    request = response.wsgi_request
+    assert LANGUAGE_SESSION_KEY not in request.session
+    assert get_lang_from_session(request, SUPPORTED_LANGUAGES) is None
+
+    # Test session with supported language.
+    response = client.get("")
+    request = response.wsgi_request
+    request.session[LANGUAGE_SESSION_KEY] = 'gl'
+    assert get_lang_from_session(request, SUPPORTED_LANGUAGES) == 'gl'
+
+    # Test session with supported language.
+    response = client.get("")
+    request = response.wsgi_request
+    request.session[LANGUAGE_SESSION_KEY] = 'en-US'
+    assert get_lang_from_session(request, SUPPORTED_LANGUAGES) == 'en-US'
+
+    # Test cookie with longer underscore language code for a supported
+    # language.
+    response = client.get("")
+    request = response.wsgi_request
+    request.session[LANGUAGE_SESSION_KEY] = 'gl_ES'
```

Comment:
missing assert ?

Reply:
No, here we are telling the session to use this particular language. Maybe there is a better way to achieve this, but this works.

Reply:
It results that the missing assert is a few lines below.

Reply:
Fixed now.